File: DesignPatternsJava/src/com/visitor/SoftDrink.java
package com.visitor;

public class SoftDrink implements ITaxable{
	private double price;
	public SoftDrink(double aPRice){
		price=aPRice;
	}public double getPrice(){
		return price;
	}public double addTaxMonitor(ITaxMonitor taxMonitor) {		
		return taxMonitor.addTax(this);
	}
}

File: DesignPatternsDotNet/VisitorPattern/SoftDrink.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VisitorPattern
{
    class SoftDrink
    { 
        private double price;    
        public SoftDrink(double aPRice)
        {
            price = aPRice;
        }
        public double Price
        {
            get
            {
                return price;
            }
        }
        public double AddTaxMonitor(ITaxMonitor taxMonitor)
        {
            return taxMonitor.AddTax(this);
        }
    }
}
